src/platform/ESP32/BUILD.gn

```diff
@@ -14,6 +14,7 @@
 
 import("//build_overrides/chip.gni")
 
+import("${chip_root}/build/chip/chip_codegen.gni")
```

Comment:
This file does no codegen though - could you explain the codepaths of why this must load the file and why we should not just rely that chip_codegen.gni is imported by builds when needed?

Reply:
See comments in #27636

apparently the failure is due to `chip_build_tests` in args, where it seems our tests depend on codegen. This is based on `CONFIG_BUILD_CHIP_TESTS` which is set to `y` in all our sdkconfig defaults.

Reply:
We may want instead to make `config/esp32/components/chip/CMakeLists.txt` not pass in `--fail-on-unused-args ` or somehow conditionall detect if the pregen_dir is required.

Reply:
> We may want instead to make `config/esp32/components/chip/CMakeLists.txt` not pass in `--fail-on-unused-args ` or somehow conditionall detect if the pregen_dir is required.

I can pass `--fail-on-unused-args` conditionally based on `CHIP_CODEGEN_PREGEN_DIR`

Reply:
I guess ... maybe with a comment explaining it is the only edge case we are aware of. Unfortunate to have it.

Reply:
Done and added the comment